app/assets/javascripts/lib/views/d3_chart_view.coffee

```diff
@@ -115,30 +115,42 @@ class @D3ChartView extends BaseChartView
     opts.left_margin     = opts.left_margin || 10
     legend_item_width    = (opts.width - opts.left_margin) / opts.columns
 
+    series = opts.series.reverse().chunk(opts.columns)
+
     legend = d3.select(@container_selector())
       .append('div')
       .attr("class", "legend")
       .style("margin-left", "#{ opts.left_margin + @margins.left }px")
-      .selectAll(".legend-item")
-      .data(opts.series)
+      .selectAll(".legend-column")
+      .data(series)
       .enter()
       .append("div")
-      .attr("class", (d) ->
-        ["legend-item",
-          (if d.get('skip') then 'hidden' else ''),
-          (if d.get('is_target_line') then 'target_line' else '')
-        ].join(' ')
-      )
+      .attr("class", "legend-column")
       .style('width', "#{ legend_item_width }px")
-      .on('click', @legend_click)
-
-    legend.append("span")
-      .attr('class', 'rect')
-      .style("background-color", (d) => d.get 'color')
 
-    legend.append("span")
-      .text((d) ->
-        d.get('label') || I18n.t("output_element_series.#{d.get('key')}")
+    d3.selectAll(".legend-column")
+      .each((items) ->
+        scope = d3.select(this)
+
+        items.forEach((d) ->
+          item = scope.append('div')
+            .attr("class", () ->
+              ["legend-item",
+                (if d.get('skip') then 'hidden' else ''),
+                (if d.get('is_target_line') then 'target_line' else '')
+              ].join(' ')
+            )
+            .on('click', @legend_click)
+
+          item.append("span")
+            .attr('class', 'rect')
+            .style("background-color", () => d.get 'color')
```

Comment:
I don't think `d.get('color')` needs to be wrapped in a function. Same for `text/label` below.